engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -467,18 +499,16 @@ case class WorkflowExecutionActor(params: WorkflowExecutionActorParams)
     }
 
     (for {
-      backendName <- workflowDescriptor
-        .backendAssignments.get(key.call).toValidNel(s"Cannot find an assigned backend for call ${key.call.fullyQualifiedName}").toEither
-      backendFactory <- backendFactories.get(backendName).toValidNel(s"Cannot find a backend factory for backend $backendName").toEither
+      factory <- backendFactoryForTaskCallNode(key.call)
       command <- runCommand
-    } yield startEJEA(key, backendName, backendFactory, command)).toValidated
+    } yield startEJEA(key, factory, command)).toValidated
   }
 
   private def startEJEA(jobKey: BackendJobDescriptorKey,
-                        backendName: String,
                         backendFactory: BackendLifecycleActorFactory,
                         command: BackendJobExecutionActorCommand): WorkflowExecutionDiff = {
     val ejeaName = s"${workflowDescriptor.id}-EngineJobExecutionActor-${jobKey.tag}"
+    val backendName = backendFactory.name
```

Comment:
Just want to check: this is valid if there are two instances of the same backend active at once, right?

Reply:
The code claims the name corresponds to the name in the config.